_posts/2017-09-05-planning-for-7.0.md

```diff
@@ -0,0 +1,610 @@
+
+Many open source projects (including Babel still 😝), tutorials, conference talks, etc all use `stage-0`. React promotes the use of JSX, class properties (now Stage 3), object rest/spread (now Stage 3) and we all believe that it's just JavaScript because Babel compiled it for them. So maybe removing this abstraction would help people understand more about what is going on and the tradeoffs one is making when choosing to use Stage X plugins.
+
+It also seems much easier to maintain your own preset than to have to update the Stage preset.
+
+> I often see people go "I want object rest, and that's stage 2, so I enabled stage 2". They now have a load of other experimental features enabled they might not know about and probably don't need.
+> Also, as stages change over time then people who aren't using shrinkwrap or yarn will get new features appearing, possibly without their knowledge. If a feature is canned they might even get one vanishing. @glenjamin
+
+### Using npm Scoped Packages
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Thoughts on <a href="https://twitter.com/babeljs">@babeljs</a> using npm scoped packages for 7.0?</p>&mdash; Henry Zhu (@left_pad) <a href="https://twitter.com/left_pad/status/821551189166878722">January 18, 2017</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Seems like most who understood what scoped packages are were in favor?
+
+Pros
+
+- Don't have to worry about getting a certain package name (the reason why this was brought up in the first place).
+
+> Many package names have been taken (preset-es2016, preset-es2017, 2020, 2040, etc). Can always ask to transfer but not always easy to do and can lead to users believing certain packages are official due to the naming.
+
+Cons
+
+- We need to migrate to new syntax
+- Still unsupported on certain non-npm tools (lock-in)
+- No download counts unless we alias back to old names
+
+Sounds like we may want to defer, and in the very least it's not a breaking change given it's a name change.
+
+### `external-helpers`, `transform-runtime`, `babel-polyfill`
+
+> "regeneratorRuntime is not defined" - reported all the time.
+
+Basically we need a better solution around how to deal with built-ins/polyfills.
+
+- Developers don't know what regenerator-runtime is, they just want to use generators/async functions.
+- Many developers are confused as to why a runtime is needed at all or why Babel doesn't compile `Promise`, `Object.assign`, or some other built-in.
+- Developers are confused with the difference between `transform-runtime` the Babel plugin and the runtime itself, `babel-runtime`.
+- Complaints about generated code size since `babel-polyfill` includes all polyfills (although now we have [`useBuiltIns`](https://github.com/babel/babel-preset-env#usebuiltins)) and no one knowing about `external-helpers`
```

Comment:
This is indeed a big issue. Anything that can be done about **default** code size would be awesome.

Reply:
Raising consciousness about `external-helpers` would be great. Such lines as [this one](https://github.com/rollup/rollup-plugin-babel/blob/a5f16de276c9a55bf6989797c2d6cda46201d5ed/src/index.js#L81) do a great job at that, same thing could happen in webpack's loader - this is the reason why I've created [this](https://github.com/babel/babel/pull/5916).

Also I was wondering if there could be a better story about sharing helpers between libraries - now its a lot of duplicated code in the community's codebases.